### PR TITLE
build/tools/esp32: Add erase read functionality

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
this command is used to readback flash contents
in esp32 boards.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [4282a9d3e19688961760f50be7b3a4cb8c7efb43] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4282a9d3e19688961760f50be7b3a4cb8c7efb43] - Code Rule Check (C++) OK.

File: build/tools/esp32/esptool_py/Makefile [hunk 1]
```diff
@@ -125,11 +128,15 @@ COMPONENT_SUBMODULES += $(COMPONENT_PATH)/esptool
 BOOTLOADER_OFFSET := 0x1000
 BOOTLOADER_BIN := $(CURRENTDIR)/bootloader/bootloader.bin
 
-
 erase_flash: $(ESPTOOLPY_SUBMODULE)
 	@echo "Erasing entire flash..."
 	$(ESPTOOLPY_SERIAL) erase_flash
 
+dump_flash: $(ESPTOOLPY_SUBMODULE)
+	    @echo "Erasing entire flash..."
```
Comment (chanijjani) [inline build/tools/esp32/esptool_py/Makefile]: Please fix this message properly...

**Comment by sunghan-chang** [comment]
erase read ?

**Comment by seinfra** [comment]
Target : [8b5cfa5888752fb4d015012a94b5071f9be664fc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8b5cfa5888752fb4d015012a94b5071f9be664fc] - Code Rule Check (C++) OK.

**Pull request merged**